### ISSUE TITLE
Quotes around gradlewRoot.name in downloadFromRemote task are counted as part of folder name

### ISSUE THREAD

**Issue by maxdroz** [skrugly/mirakle#106]
I run Miracle on Mac OS and my remote server is running Linux
`rsync` version 3.2.4

I get
`rsync: change_dir "/home/remote-user/mainframer/"my-app"" failed: No such file or directory (2)` 
error when running gradle build.

I checked, folder `/home/remote-user/mainframer/my-app` did exist on remote machine.

It turns out, quotes here are a part of folder name. 
Miracle creates folder `my-app`, executes in folder `my-app`, but tries to download from folder `"my-app"`

I have no idea, how this has not been caught yet, so I guess it has something to do with this specific OS combination.

P.S. I did a workaround by creating a symlink from folder `my-app` to `"my-app"` and it started to work fine.

**Comment by Virtlink** [comment]
I have the same issue since I updated rsync on my Mac from the default version 2.6.9 to Homebrew's rsync 3.2.4.

This is my script:
```gradle
initscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath "io.github.adambl4:mirakle:1.5.0"
    }
}

apply plugin: Mirakle

rootProject {
    mirakle {
        host "builder"

        remoteFolder ".mirakle"

        rsyncToRemoteArgs += ["--archive", "--no-perms", "--compress", "--stats", "--human-readable"]
        rsyncFromRemoteArgs += ["--archive", "--no-perms", "--compress", "--stats", "--human-readable"]

        downloadInParallel true
        downloadInterval 2000
    }
}
```

With rsync 2, the output is:

```
> Configure project :
Here's Mirakle 1.5.0. All tasks will be executed on builder.

> Task :uploadToRemote

Number of files: 91965
Number of files transferred: 0
Total file size: 1.17G bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 3686639
File list generation time: 0.748 seconds
File list transfer time: 0.000 seconds
Total bytes sent: 3.69M
Total bytes received: 26

sent 3.69M bytes  received 26 bytes  819.30K bytes/sec
total size is 1.17G  speedup is 317.12

> Task :executeOnRemote

FAILURE: Build completed with 2 failures.

BUILD FAILED in 1s

> Task :downloadInParallel

Number of files: 82720
Number of files transferred: 0
Total file size: 1.09G bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 3448171
File list generation time: 0.647 seconds
File list transfer time: 0.000 seconds
Total bytes sent: 172
Total bytes received: 3.45M

sent 172 bytes  received 3.45M bytes  985.31K bytes/sec
total size is 1.09G  speedup is 315.06

> Task :downloadFromRemote

Number of files: 82720
Number of files transferred: 0
Total file size: 1.09G bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 3448171
File list generation time: 0.654 seconds
File list transfer time: 0.000 seconds
Total bytes sent: 172
Total bytes received: 3.45M

sent 172 bytes  received 3.45M bytes  985.31K bytes/sec
total size is 1.09G  speedup is 315.06

> Task :mirakle FAILED
Total time : 12.622 secs
Task uploadToRemote took : 4.064 secs
Task executeOnRemote took : 2.449 secs
Task downloadFromRemote took : 3.08 secs
5 actionable tasks: 5 executed
```

This is the output when using rsync 3:
```
> Configure project :
Here's Mirakle 1.5.0. All tasks will be executed on builder.

> Task :uploadToRemote

Number of files: 91,965 (reg: 86,092, dir: 5,873)
Number of created files: 0
Number of deleted files: 0
Number of regular files transferred: 0
Total file size: 1.17G bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 2.57M
File list generation time: 0.002 seconds
File list transfer time: 0.000 seconds
Total bytes sent: 4.15M
Total bytes received: 6.35K

sent 4.15M bytes  received 6.35K bytes  923.00K bytes/sec
total size is 1.17G  speedup is 281.49

> Task :executeOnRemote

FAILURE: Build completed with 2 failures.

BUILD FAILED in 970ms

> Task :downloadInParallel
rsync: change_dir "/home/username//.mirakle/"devenv"" failed: No such file or directory (2)

Number of files: 0
Number of created files: 0
Number of deleted files: 0
Number of regular files transferred: 0
Total file size: 0 bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 105
Total bytes sent: 164
Total bytes received: 105

sent 164 bytes  received 105 bytes  107.60 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1855) [Receiver=3.2.4]
rsync: [Receiver] write error: Broken pipe (32)
Parallel download failed with exception org.gradle.process.internal.ExecException: Process 'command 'rsync'' finished with non-zero exit value 23

> Task :downloadFromRemote FAILED
rsync: change_dir "/home/username//.mirakle/"devenv"" failed: No such file or directory (2)

Number of files: 0
Number of created files: 0
Number of deleted files: 0
Number of regular files transferred: 0
Total file size: 0 bytes
Total transferred file size: 0 bytes
Literal data: 0 bytes
Matched data: 0 bytes
File list size: 105
Total bytes sent: 164
Total bytes received: 105

sent 164 bytes  received 105 bytes  179.33 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1855) [Receiver=3.2.4]
rsync: [Receiver] write error: Broken pipe (32)
Total time : 9.19 secs
Task uploadToRemote took : 4.073 secs
Task executeOnRemote took : 2.37 secs
Task downloadFromRemote took : 1.384 secs

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':downloadFromRemote'.
> Process 'command 'rsync'' finished with non-zero exit value 23

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.9.2/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 9s
4 actionable tasks: 4 executed
```

**Comment by KilowattJunkie** [comment]
I can confirm this, and am using rsync 3, but it also appears that the quotes may be being added manually when [creating the download path](https://github.com/Adambl4/mirakle/blob/f274342842e361b46524f9ae6ae544fd1d794619/plugin/src/main/kotlin/Mirakle.kt#L182)

`"${config.host}:${config.remoteFolder}/\"${gradlewRoot.name}\"/",`

However, I'm not sure what the implications of removing the `\"` are, if this was working before...

**Comment by skrugly** [comment]
Fixed in 1.5.1
